api.go

```diff
@@ -159,11 +166,21 @@ func (api *CacophonyAPI) UploadThermalRaw(info *cptvInfo, r io.Reader) error {
 	req.Header.Set("Authorization", api.token)
 
 	client := new(http.Client)
+	client.Timeout = time.Duration(timeout) * time.Second
 	resp, err := client.Do(req)
 	if err != nil {
 		return err
 	}
-	resp.Body.Close()
+	if resp.StatusCode != 200 {
+		bodyBytes, err := ioutil.ReadAll(resp.Body)
+		if err != nil {
+			return err
+		}
+		bodyString := string(bodyBytes)
+		resp.Body.Close()
```

Comment:
The body always has to be closed, even if it isn't used or you end up leaking memory and perhaps file handles. Better put a `defer` before the `if`.